src/main.zig

```diff
@@ -2704,9 +2704,22 @@ pub fn cmdFmt(gpa: *Allocator, args: []const []const u8) !void {
             fatal("cannot use --stdin with positional arguments", .{});
         }
 
-        const stdin = io.getStdIn().reader();
-
-        const source_code = try stdin.readAllAlloc(gpa, max_src_size);
+        const stdin = io.getStdIn();
+
+        const source_code = blk: {
+            const source_code = try stdin.readToEndAllocOptions(gpa, max_src_size, null, @alignOf(u16), null);
+            errdefer gpa.free(source_code);
+
+            // If the file starts with a UTF-16 BOM, translate it to UTF-8
+            if (mem.eql(u8, source_code[0..2], "\xff\xfe")) {
```

Comment:
```suggestion
            if (mem.startsWith(u8, source_code, "\xff\xfe")) {
```

This handles the case where `source_code.len < 2` without erroring.